verify.go

```diff
@@ -276,6 +279,32 @@ func (v *IDTokenVerifier) Verify(ctx context.Context, rawIDToken string) (*IDTok
 		}
 	}
 
+	// If nbf claim is provided in token, ensure that it is indeed in the past.
+	if !v.config.SkipNbfCheck {
+		var claimsMap map[string]interface{}
+		err := json.Unmarshal(t.claims, &claimsMap)
+		if err != nil {
+			return nil, fmt.Errorf("oidc: failed to unmarshal claims: %v", err)
+		}
+		if nbfRaw := claimsMap["nbf"]; nbfRaw != nil {
+			nbf, ok := nbfRaw.(float64)
```

Comment:
Add NotBeforeField to IDToken starting from here:

https://github.com/coreos/go-oidc/blob/8ae1da518bd4d9d5a5909090a184af30f336436d/oidc.go#L323

since it's a public claim.

Reply:
My rationale was that the nbf claim is not [ID Token specific](https://openid.net/specs/openid-connect-core-1_0.html#IDToken) but a more ["common" claim](https://tools.ietf.org/html/rfc7519#section-4.1.5) which none of the other ones listed so far is - perhaps there is some value of keeping a distinction between the two? If not I'll add a NotBeforeField as you suggest.

Reply:
That struct isn't exported, it's solely used for this method and validation :) Feel free to add any fields to it.

You can also use a pointer to determine presence

```go
type idToken struct {
    IssuedAt  jsonTime  `json:"iat"`
    NotBefore *jsonTime `json:"nbf"`
}

// ...

var t idToken
// ... unmarshal

if t.NotBefore != nil {
    // "nbf" is present. Evaluate it
}
```

Reply:
Nice one! Thanks 👍 Check looking so much simpler now.